features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/net/ResourceRequestBodyFactory.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.net
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.storage.RawBatchEvent
+import com.datadog.android.sessionreplay.internal.exception.InvalidPayloadFormatException
+import com.datadog.android.sessionreplay.internal.utils.MiscUtils
+import com.google.gson.JsonObject
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import kotlin.jvm.Throws
+
+internal class ResourceRequestBodyFactory(
+    private val internalLogger: InternalLogger
+) {
+
+    @Throws(InvalidPayloadFormatException::class)
+    internal fun create(
+        resources: List<RawBatchEvent>
+    ): RequestBody {
+        val deserializedResources: List<ResourceEvent> = resources.mapNotNull {
+            val resourceMetadata = MiscUtils.safeDeserializeToJsonObject(internalLogger, it.metadata)
+
+            if (resourceMetadata != null) {
+                val applicationId = MiscUtils.safeGetStringFromJsonObject(
+                    internalLogger,
+                    resourceMetadata,
+                    APPLICATION_ID_KEY
+                )
+                val filename = MiscUtils.safeGetStringFromJsonObject(
+                    internalLogger,
+                    resourceMetadata,
+                    FILENAME_KEY
+                )
+
+                if (applicationId != null && filename != null) {
+                    ResourceEvent(
+                        applicationId = applicationId,
+                        identifier = filename,
+                        it.data
+                    )
+                } else {
+                    null
+                }
+            } else {
+                null
+            }
+        }
+
+        if (deserializedResources.isEmpty()) {
+            @Suppress("ThrowingInternalException")
+            throw InvalidPayloadFormatException(NO_RESOURCES_TO_SEND_ERROR)
+        }
```

Comment:
> [!WARNING]
> Please let's not throw exceptions here. We're not implementing an interface so we can decide what the signature is, it's better to return a null here than throw an exception. 
> 
> First, creating an exception means that the VM needs to build the stack trace which is costly.
> Second, even with the `@Throws` annotation, there's no checked exceptions in Kotlin, meaning we don't have a way to enforce that the caller use a `try/catch` block, which could lead to us crashing the host app.

Reply:
In continuation of what we discussed, I've modified DataOkHttpUploader to accept a nullable request and giving an error upload status if null is received